shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -1126,6 +1126,19 @@ public boolean shouldRestoreAndSaveState() {
     return true;
   }
 
+  /**
+   * Give the host application a chance to take control of the app lifecycle events.
+   *
+   * <p>Return {@code false} means the host application dispatches these app lifecycle events, while
+   * return {@code true} means the engine dispatches these events.
```

Comment:
can we explain why anyone may want to do this? The more context, the better. 

I left some areas to cover in https://github.com/flutter/flutter/issues/94702#issuecomment-1018000395

Reply:
> can we explain why anyone may want to do this? The more context, the better.
> 
> I left some areas to cover in [flutter/flutter#94702 (comment)](https://github.com/flutter/flutter/issues/94702#issuecomment-1018000395)

I have added more context in the interface description.

Reply:
done.